test/e2e/aks_fleets_member.go

```diff
@@ -0,0 +1,152 @@
+		Tags: map[string]*string{
+			"jobName":           ptr.To(os.Getenv(JobName)),
+			"creationTimestamp": ptr.To(os.Getenv(Timestamp)),
+		},
+	}, nil)
+	Expect(err).NotTo(HaveOccurred())
+
+	fleetClient, err := armcontainerservicefleet.NewFleetsClient(getSubscriptionID(Default), cred, nil)
+	Expect(err).NotTo(HaveOccurred())
+
+	fleetsMemberClient, err := armcontainerservicefleet.NewFleetMembersClient(getSubscriptionID(Default), cred, nil)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Creating a fleet manager")
+	poller, err := fleetClient.BeginCreateOrUpdate(ctx, groupName, fleetName, armcontainerservicefleet.Fleet{
+		Location: ptr.To(os.Getenv(AzureLocation)),
+	}, nil)
+	Expect(err).NotTo(HaveOccurred())
+	_, err = poller.PollUntilDone(ctx, nil)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Joining the cluster to the fleet hub")
+	var infraControlPlane = &infrav1.AzureManagedControlPlane{}
+	Eventually(func(g Gomega) {
+		err = mgmtClient.Get(ctx, client.ObjectKey{Namespace: input.Cluster.Spec.ControlPlaneRef.Namespace, Name: input.Cluster.Spec.ControlPlaneRef.Name}, infraControlPlane)
+		g.Expect(err).NotTo(HaveOccurred())
+		infraControlPlane.Spec.FleetsMember = &infrav1.FleetsMember{
+			FleetsMemberClassSpec: infrav1.FleetsMemberClassSpec{
+				ManagerName:          fleetName,
+				ManagerResourceGroup: groupName,
+				Group:                updateGroupName,
+			},
+		}
+		g.Expect(mgmtClient.Update(ctx, infraControlPlane)).To(Succeed())
+		g.Expect(conditions.IsTrue(infraControlPlane, infrav1.FleetReadyCondition)).To(BeTrue())
+	}, input.WaitIntervals...).Should(Succeed())
+
+	By("Ensuring the fleet member is created and attached to the managed cluster")
+	Eventually(func(g Gomega) {
+		resp, err := fleetsMemberClient.Get(ctx, groupName, fleetName, input.Cluster.Name, nil)
+		g.Expect(err).NotTo(HaveOccurred())
+		g.Expect(resp.Properties.ProvisioningState).To(Equal(ptr.To(armcontainerservicefleet.FleetMemberProvisioningStateSucceeded)))
+		fleetsMember := resp.FleetMember
+		g.Expect(fleetsMember.Properties).NotTo(BeNil())
+		expectedID := azure.ManagedClusterID(getSubscriptionID(Default), infraControlPlane.Spec.ResourceGroupName, input.Cluster.Name)
+		g.Expect(fleetsMember.Properties.ClusterResourceID).To(Equal(ptr.To(expectedID)))
+		g.Expect(fleetsMember.Properties.ProvisioningState).To(Equal(ptr.To(armcontainerservicefleet.FleetMemberProvisioningStateSucceeded)))
+	}, input.WaitIntervals...).Should(Succeed())
+
+	By("Remove the FleetsMember spec from the AzureManagedControlPlane")
+	Eventually(func(g Gomega) {
+		err = mgmtClient.Get(ctx, client.ObjectKey{Namespace: input.Cluster.Spec.ControlPlaneRef.Namespace, Name: input.Cluster.Spec.ControlPlaneRef.Name}, infraControlPlane)
+		g.Expect(err).NotTo(HaveOccurred())
+		infraControlPlane.Spec.FleetsMember = nil
+		g.Expect(mgmtClient.Update(ctx, infraControlPlane)).To(Succeed())
+	}, input.WaitIntervals...).Should(Succeed())
+
+	By("Waiting for the managed cluster to finish updating")
```

Comment:
Not blocking this PR, but I'm still curious as to exactly what's being updated here if removing the fleets member field doesn't actually delete the ASO resource.

Reply:
I think that there was an issue with the controller trying to reconcile the fleets member as we're trying to delete it, and it may have been causing some problems. I'll try testing it again without this step and maybe those problems were due to a different issue.

Reply:
Found the error. It happens when I try to delete the fleets member:
```
this operation depends on is not in an expected state. Expected: ProvisioningState in (Succeeded, Canceled, Failed, Deleting, Upgrading). Actual: ProvisioningState is Updating. Change the resource to expected state and try again.
```